fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java

```diff
@@ -236,4 +200,103 @@ private void updatePartitionedStats(
+
+        Set<Integer> identityPartitionIds = IcebergUtil.getIdentityPartitions(icebergTable.spec()).keySet().stream()
+                .map(PartitionField::sourceId)
+                .collect(toSet());
+
+        List<Types.NestedField> nonPartitionPrimitiveColumns = columns.stream()
+                .filter(column -> !identityPartitionIds.contains(column.fieldId()) && column.type().isPrimitiveType())
+                .collect(toImmutableList());
+
+        TableScan tableScan = IcebergUtil.getTableScan(icebergTable,
+                snapshot.get(), icebergPredicates, true);
+
+        IcebergFileStats icebergFileStats = null;
+        try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
+            for (FileScanTask fileScanTask : fileScanTasks) {
+                DataFile dataFile = fileScanTask.file();
+                if (icebergFileStats == null) {
+                    icebergFileStats = new IcebergFileStats(
+                            idToTypeMapping,
+                            nonPartitionPrimitiveColumns,
+                            dataFile.partition(),
+                            dataFile.recordCount(),
+                            dataFile.fileSizeInBytes(),
+                            IcebergFileStats.toMap(idToTypeMapping, dataFile.lowerBounds()),
+                            IcebergFileStats.toMap(idToTypeMapping, dataFile.upperBounds()),
+                            dataFile.nullValueCounts(),
+                            dataFile.columnSizes());
+                } else {
+                    icebergFileStats.incrementFileCount();
+                    icebergFileStats.incrementRecordCount(dataFile.recordCount());
+                    icebergFileStats.incrementSize(dataFile.fileSizeInBytes());
+                    updateSummaryMin(icebergFileStats, partitionFields, IcebergFileStats.toMap(idToTypeMapping,
+                            dataFile.lowerBounds()), dataFile.nullValueCounts(), dataFile.recordCount());
+                    updateSummaryMax(icebergFileStats, partitionFields, IcebergFileStats.toMap(idToTypeMapping,
+                            dataFile.upperBounds()), dataFile.nullValueCounts(), dataFile.recordCount());
+                    icebergFileStats.updateNullCount(dataFile.nullValueCounts());
+                    updateColumnSizes(icebergFileStats, dataFile.columnSizes());
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        return icebergFileStats;
+    }
+
+    private boolean generateColumnStatistic(IcebergFileStats icebergFileStats,
+                                            ColumnStatistic.Builder columnBuilder,
+                                            int fieldId,
+                                            double recordCount,
+                                            List<ColumnRefOperator> columnList) {
+        boolean isEstimated = true;
```

Comment:
How about return ColumnStatistic instread of boolean, you  can return ColumnStatistic::UNKNOWN directly if isEstimated os false, and check isUnknown outside